packages/amazonq/test/e2e/amazonq/transformByQ.test.ts

```diff
@@ -0,0 +1,270 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import assert from 'assert'
+import { qTestingFramework } from './framework/framework'
+import sinon from 'sinon'
+import { Messenger } from './framework/messenger'
+import { JDKVersion } from 'aws-core-vscode/codewhisperer'
+import { GumbyController, startTransformByQ, TabsStorage } from 'aws-core-vscode/amazonqGumby'
+
+describe('Amazon Q Code Transformation', function () {
+    let framework: qTestingFramework
+    let tab: Messenger
+
+    beforeEach(() => {
+        framework = new qTestingFramework('gumby', true, [])
+        tab = framework.createTab()
+    })
+
+    afterEach(() => {
+        framework.removeTab(tab.tabID)
+        framework.dispose()
+        sinon.restore()
+    })
+
+    describe('Quick action availability', () => {
+        it('Shows /transform when QCT is enabled', async () => {
+            const command = tab.findCommand('/transform')
+            if (!command) {
+                assert.fail('Could not find command')
+            }
+
+            if (command.length > 1) {
+                assert.fail('Found too many commands with the name /transform')
+            }
+        })
+
+        it('Does NOT show /transform when QCT is NOT enabled', () => {
+            framework.dispose()
+            framework = new qTestingFramework('gumby', false, [])
+            const tab = framework.createTab()
+            const command = tab.findCommand('/transform')
+            if (command.length > 0) {
+                assert.fail('Found command when it should not have been found')
+            }
+        })
+    })
+
+    describe('Starting a transformation from chat', () => {
+        it('Can click through all user input forms for a Java upgrade', async () => {
+            sinon.stub(startTransformByQ, 'getValidSQLConversionCandidateProjects').resolves([])
```

Comment:
I stub some validation-related logic in this test file because 1) we already have unit test coverage of those, and 2) these tests are just as useful even with mocked validation functions.

Reply:
Ok, but when things change in the implementation (e.g. using Flare), stubs/mocks will break easily. Thanks for calling this out in any case.